library/cc/log_level.cc

```diff
@@ -0,0 +1,31 @@
+#include "log_level.h"
+
+// NOLINT(namespace-envoy)
+
+#include <stdexcept>
+
+static const std::pair<LogLevel, std::string> LOG_LEVEL_LOOKUP[]{
+    {LogLevel::trace, "trace"}, {LogLevel::debug, "debug"}, {LogLevel::info, "info"},
+    {LogLevel::warn, "warn"},   {LogLevel::error, "error"}, {LogLevel::critical, "critical"},
+    {LogLevel::off, "off"},
+};
```

Comment:
is there a canonical solution to mapping between `enum` and `std::string` in C++? this kind of lookup table (which i use in several places) is what i came up with

Reply:
looking around on the internet, seems like there isn't a great approach to mapping an enum to an arbitrary string without depending on compiler-specific internals. going to just resolve this.